HULKs/hulk
# Transition Interpolator

## Introduced Changes

Adds a `TransitionInterpolator` which interpolates between only two keys, replacing `LinearInterpolator`.
For the `dispatching_interpolator`, the hard coded value of `1s` is removed and replaced by a calculation of
transition time using the `maximum_joint_velocities` (which is a new parameter).
This theoretically allows for a faster transition if the robot is already close to the target position.

Fixes #

## ToDo / Known Issues
* The `maximum_joint_velocities` are chosen pretty arbitrarily

## Ideas for Next Iterations (Not This PR)

*If there are some improvements that could be done in a next iteration, describe them here.*

## How to Test

* Build and upload
* A new parameter `maximum_joint_velocities` is available
* The dispatching interpolator movements should look the same, the duration can be different however
* The arm movement while walking should be unchanged
